CDCgov/prime-reportstream
# Implement CLI update utility for observation mapping table

This PR implements a CLI command for updating the observation mapping table.

Test Steps:
1. Run the unit tests
2. Grab a license for the NLMS VSAC
    1. Visit the [VSAC Support Center](https://www.nlm.nih.gov/vsac/support/index.html)
    2. Click on `Request a UMLS License` under `How to Use VSAC`
    3. Login with an identity provider; I used Google
    4. Get apikey from profile to populate parameter `-k` in commands below
3. Manually test the CLI utility
    1. Download [special sample mapping table](https://github.com/CDCgov/prime-reportstream/files/13442875/observation-mapping-to-update.csv)
    2. Load the table 
    ```zsh
    ./prime lookuptables create -i observation-mapping-to-update.csv -a -n observation-mapping -s -f
    ``` 
    3. Run an update; be sure to select yes when prompted to write the table or output a new local csv
    ```zsh
    ./prime lookuptables update-mapping -a -n observation-mapping -k YOUR_API_KEY -o out.csv
    ```
    4. Verify
        1. That there are new/updated rows for all entries with OID values
            1. Should have 5 rows for `OID` `2.16.840.1.113762.1.4.1146.828`
            2. Should have 2 rows for `OID` `2.16.840.1.113762.1.4.1146.1112`
            3. All `SHOULD CHANGE` values have changed
        2. Condition values (`condition_name`, `condition_code`, `Condition Code System`, `Condition Code System Version`, `Created At`) should have all remained the same
        3. `SHOULD REMAIN` values have stayed (same number of rows)
        4. `out.csv` was written with the correct values
        5. `metadata/tables/local/observation-mapping.csv` was written with the correct values
4. Repeat step 2 but specify the OID to update
    1. Run an update specifying an OID
    ```zsh
    ./prime lookuptables update-mapping -a -n observation-mapping -k YOUR_API_KEY -d 2.16.840.1.113762.1.4.1146.828
    ```
    2. Verify
        1. That there are new/updated rows for the specified OID value
            1. Should have 5 rows for `OID` `2.16.840.1.113762.1.4.1146.828`
            2. All `SHOULD CHANGE` values **_for OID above_** have changed
            3. All `SHOULD CHANGE` values for other OID **_have not changed_**
        2. Condition values (`condition_name`, `condition_code`, `Condition Code System`, `Condition Code System Version`, `Created At`) should have all remained the same
        3. `SHOULD REMAIN` values have stayed (same number of rows)

## Changes
- Implemented the CLI utility `lookuptables update-mapping` for updating observation mappings using a ValueAuthority
- Adds a utility class for holding pertinent constants

## Checklist

### Testing

## Linked Issues
- Fixes #11510
